Issue 44110 - Support for displaying run object inputs and object outputs

ankurjuneja opened the pull request:
#### Rationale
This PR provides support to register run inputs and run outputs views with experiment service and display those views in run details view.

#### Related Pull Requests
* https://github.com/LabKey/provenance/pull/82

#### Changes
* View registration support

ankurjuneja committed:
display object inputs and outputs on run details page

ankurjuneja committed:
show if view is not empty

labkey-jeckels commented on experiment/src/org/labkey/experiment/api/ExperimentServiceImpl.java:
```diff
@@ -7710,6 +7714,30 @@ public boolean useUXDomainDesigner()
+    @Override
+    public Set<QueryForm> getRunInputsQueries()
+    {
+        return _runInputsQueryForms;
```
Please return an immutable wrapper over these collections via Collections.unmodifiableSet()

ankurjuneja replied:
done

labkey-jeckels commented on api/src/org/labkey/api/exp/api/ExperimentService.java:
```diff
@@ -792,6 +793,14 @@ ExpRun derive(Map<ExpMaterial, String> inputMaterials, Map<ExpData, String> inpu
+
+    Set<QueryForm> getRunOutputsQueries();
+
+    void registerRunInputsViewProvider(Set<QueryForm> queryForms);
```
A comment here about where this is used and the expectation that the referenced query has a RunId column would be helpful.

ankurjuneja replied:
done

labkey-jeckels reviewed (approved):
Some recommendations but no need for additional review.

ankurjuneja committed:
code review feedback changes

labkey-jeckels committed:
Fix view creation script for Postgres
Make view creation consistent for SQLServer
Switch to a QueryViewProvider approach to avoid thread-unsafe modification of QueryForms showing provenance data

ankurjuneja committed:
Merge branch 'release21.11-SNAPSHOT' into 21.11_fb_44110_objectIPOP

ankurjuneja committed:
Merge branch 'release21.11-SNAPSHOT' into 21.11_fb_44110_objectIPOP
